lib/archethic/mining/validation_context.ex

```diff
@@ -1026,7 +1025,7 @@ defmodule Archethic.Mining.ValidationContext do
 
       _ ->
         Transaction.verify_origin_signature?(tx, pow) and
-          SharedSecrets.has_origin_public_key?(pow)
+          pow in ProofOfWork.list_origin_public_keys_candidates(tx)
```

Comment:
unit tests

Reply:
There is already tests for this check: https://github.com/archethic-foundation/archethic-node/blob/0aa05779dcca7455ffe1f3c688e541a10c2666f0/test/archethic/mining/validation_context_test.exs#L75